src/uct/base/uct_iface.c

```diff
@@ -222,8 +222,17 @@ int uct_iface_is_reachable(const uct_iface_h iface, const uct_device_addr_t *dev
 int uct_iface_is_reachable_v2(const uct_iface_h iface,
                               const uct_iface_is_reachable_params_t *params)
 {
-    ucs_fatal("uct_iface_is_reachable_v2 not supported yet");
-    return 0;
+    const uct_base_iface_t *base_iface = ucs_derived_of(iface, uct_base_iface_t);
+
+    if ((params->field_mask & UCT_IFACE_IS_REACHABLE_FIELD_INFO_STRING) ||
+        (params->field_mask & UCT_IFACE_IS_REACHABLE_FIELD_INFO_STRING_LENGTH)) {
+            UCT_CHECK_PARAM((params->info_string != NULL),
+                             "info_string shouldn't be null");
+            UCT_CHECK_PARAM((params->info_string_length > 0),
+                             "info_string_length should be greater than zero");
+    }
+    
+    return base_iface->internal_ops->iface_is_reachable_v2(iface, params);
```

Comment:
need to deprecate the usage of "iface_is_reachable" field to have single flow in UCT code:
1. uct_iface_is_reachable will also call internal_ops->iface_is_reachable_v2
2. remove the implementation of iface_is_reachable callback from transports (maybe point it to ucs_empty_function_do_assert or similar). keep only the implementation of iface_is_reachable_v2

Reply:
Hi Yossi, not all components (currently only IB supports) are supported v2 api since 500 lines code change limitation. Most of them is ucs_empty_function_return_unsupported. We will supoort them in later PR. So once we support all and we will replace v1 by v2 in wireup.c and deprecate v1's API.

How about that? Thanks

Reply:
ok